java-api-core/src/com/kinvey/java/query/MongoQueryFilter.java

```diff
@@ -202,22 +203,28 @@ private Boolean negate(LinkedHashMap<String, Object> map) {
                 } else if (op != null) {
                     switch(op) {
                         case GREATERTHAN:
-                            newMap.put("$lte", obj);
+                            element.put("$gt", obj);
+                            newMap.put("$not", element);
                             break;
                         case LESSTHAN :
-                            newMap.put("$gte", obj);
+                            element.put("$lt", obj);
+                            newMap.put("$not", element);
                             break;
                         case GREATERTHANEQUAL:
-                            newMap.put("$lt", obj);
+                            element.put("$gte", obj);
+                            newMap.put("$not", element);
                             break;
                         case LESSTHANEQUAL:
-                            newMap.put("$gt", obj);
+                            element.put("$lte", obj);
+                            newMap.put("$not", element);
                             break;
                         case IN :
-                            newMap.put("$nin", obj);
+                            element.put("$in", obj);
+                            newMap.put("$not", element);
                             break;
                         case NOTIN :
-                            newMap.put("$in", obj);
+                            element.put("$nin", obj);
+                            newMap.put("$not", element);
```

Comment:
Would this effectively make this filter an `$in` filter? If so, this is not the desired effect of the `NOTIN` operator. I believe that this should still put an `$in` in the `element`, and then apply the `$not` as with the other operations. Either that, or it should just be the `$nin` operation that gets placed in `newMap`, but there may be differences in the results of both of these, and we should understand what they are.

Reply:
This changes is located in negate() method. So expected result of  negate `case:NOTIN`  - `IN`. 
Previously, we just inserted into the map `$in` Now the structure of map has changed and we need to write down the negation explicitly. We already have tests for method `$in`. We changed implementation of operation `$not` but result of this operation have to stay the same. So if old tests for operator `$not` are passed, we can think that the operator is working right